### PR TITLE
[Jobs] Add end to end test for multiple drivers in a Ray Job

### PR DESCRIPTION
**Pull request by architkulkarni**
Signed-off-by: Archit Kulkarni <architkulkarni@users.noreply.github.com>

## Why are these changes needed?
Followup for #32388 which adds an end to end test for the case of multiple drivers (`ray.init()` calls) in a single job entrypoint script.  The test fails without that PR.  

The previous PR only included a unit test for an internal function in C++.  The e2e test in this PR ensures that this use case is supported even if the internals change.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — COMMENTED** [review]
1. Should we verify that the output of the API that feeds the dashboard matches whatever is expected here? (assuming single submission ID, multiple driver IDs)
2. Should we add a test where two drivers run at the same time? (e.g., `python test.py & python test.py && fg`)

**Comment by architkulkarni** [comment]
(cc @gvspraveen )

**Comment by architkulkarni** [comment]
> 1. Should we verify that the output of the API that feeds the dashboard matches whatever is expected here? (assuming single submission ID, multiple driver IDs)
> 2. Should we add a test where two drivers run at the same time? (e.g., `python test.py & python test.py && fg`)

Good idea, let me add those tests

**Comment by architkulkarni** [comment]
@edoakes Added more tests, please take another look

File: dashboard/modules/snapshot/tests/test_job_submission.py [hunk 1]
```diff
@@ -189,5 +189,44 @@ def wait_for_job_to_fail():
     wait_for_condition(wait_for_job_to_fail, timeout=45)
 
 
+def test_multiple_ray_init(
```
Comment (edoakes) [inline dashboard/modules/snapshot/tests/test_job_submission.py:192]: please add a header comment describing the purpose of this test

**Review by edoakes — APPROVED** [review]

**Comment by architkulkarni** [comment]
Build images failure "Observed wheel commit () " known flaky on master, unrelated

**Pull request merged**